client/header/activity-panel/panels/orders.js

```diff
@@ -64,49 +63,68 @@ function OrdersPanel( { orders, isRequesting, isError } ) {
 		</EllipsisMenu>
 	);
 
-	const orderCardTitle = ( order, address ) => {
-		const name = `${ address.first_name } ${ address.last_name }`;
+	const getCustomerString = order => {
+		const { first_name, last_name } = order.extended_info.customer;
+
+		if ( ! first_name && ! last_name ) {
+			return '';
+		}
+
+		const name = [ first_name, last_name ].join( ' ' );
+		return sprintf(
+			__(
+				/* translators: describes who placed an order, e.g. Order #123 placed by John Doe */
+				'placed by {{customerLink}}%(customerName)s{{/customerLink}}',
+				'wc-admin'
+			),
+			{
+				customerName: name,
+			}
+		);
+	};
 
+	const orderCardTitle = order => {
 		return (
 			<Fragment>
 				{ interpolateComponents( {
 					mixedString: sprintf(
 						__(
-							/* eslint-disable-next-line max-len */
-							'Order {{orderLink}}#%(orderNumber)s{{/orderLink}} placed by {{customerLink}}%(customerName)s{{/customerLink}} {{destinationFlag/}}',
+							'Order {{orderLink}}#%(orderNumber)s{{/orderLink}} %(customerString)s {{destinationFlag/}}',
 							'wc-admin'
 						),
 						{
-							orderNumber: order.number,
-							customerName: name,
+							orderNumber: order.order_id,
+							customerString: getCustomerString( order ),
 						}
 					),
 					components: {
-						orderLink: <Link href={ 'post.php?action=edit&post=' + order.id } type="wp-admin" />,
+						orderLink: (
+							<Link href={ 'post.php?action=edit&post=' + order.order_id } type="wp-admin" />
+						),
+						destinationFlag: order.extended_info.customer.country ? (
+							<Flag code={ order.extended_info.customer.country } round={ false } />
+						) : null,
```

Comment:
Thinking while we are here we could hook up this customerLink to the related page in the Customers Report.

Reply:
I don't think that option is yet available to us, but I went ahead and opened an issue [here](https://github.com/woocommerce/wc-admin/issues/1594) to handle that.

In the meantime, I hooked up the link but it wont be functional until #1594 is resolved.